utils/helpers.ts

```diff
@@ -1,49 +1,56 @@
-import { DropdownOption, Plant, SeasonEnum } from '@/types/schema';
+import {
+  DropdownOption,
+  Plant,
+  PlantingTypeEnum,
+  SeasonEnum,
+  SunlightEnum,
+} from '@/types/schema';
 
-// Helper function to process late/early month fields for checkGrowingSeason
+/* Helper function to process late/early month fields
+Assumes that month is not null. 
+Assumes that the month is a valid month string; 
+otherwise, returns all-caps version of the input 
+Returns all-caps month, without the LATE_ or EARLY_ prefix
+*/
 function processPlantMonth(month: string) {
-  // If field is not null and starts with 'LATE' or 'EARLY,
-  // get substring after 'LATE_ or 'EARLY_'
-  if (!month) {
-    return month;
-  }
-
   if (month.startsWith('LATE')) {
-    return month.substring(5);
+    return month.substring(5).toLocaleUpperCase();
   } else if (month.startsWith('EARLY')) {
-    return month.substring(6);
+    return month.substring(6).toLocaleUpperCase();
   } else {
-    return month;
+    return month.toLocaleUpperCase();
   }
 }
 
+// helper constants for processing months to indexes
+const growingSeasonToIndex = new Map<SeasonEnum, number[]>([
```

Comment:
note: we might want to eventually refactor these into Records rather than Maps, since we don't need all the features of Map